File: java-hfsframework-primefaces/hfsframework-admin/src/main/java/br/com/hfsframework/admin/AdmParametroCategoriaBC.java
/**
 * <p><b>HFS Framework</b></p>
 * @author Henrique Figueiredo de Souza
 * @version 1.0
 * @since 2017
 */
package br.com.hfsframework.admin;

import br.com.hfsframework.admin.data.AdmParametroCategoriaRepository;
import br.com.hfsframework.admin.model.AdmParametroCategoria;
import br.com.hfsframework.base.BaseBusinessController;

/**
 * The Class AdmParametroCategoriaBC.
 */
public class AdmParametroCategoriaBC extends BaseBusinessController<AdmParametroCategoria, Long, AdmParametroCategoriaRepository> {

	/** The Constant serialVersionUID. */
	private static final long serialVersionUID = 1L;

	/* (non-Javadoc)
	 * @see br.com.hfsframework.base.BaseBusinessController#getDescricao(java.io.Serializable)
	 */
	@Override
	public String getDescricao(Long id){
		return repositorio.getDescricao(id);
	}
	
	/* (non-Javadoc)
	 * @see br.com.hfsframework.base.BaseBusinessController#existeNovo(java.lang.String)
	 */
	@Override
	public boolean existeNovo(String novo){
		Long existe = repositorio.countNovo(novo.toLowerCase());
		return (existe > 0);
	}
	
	/* (non-Javadoc)
	 * @see br.com.hfsframework.base.BaseBusinessController#existeAntigoNovo(java.lang.String, java.lang.String)
	 */
	@Override
	public boolean existeAntigo(Long id, String novo){
		String antigo = getDescricao(id);
		Long existe = repositorio.countAntigo(antigo.toLowerCase(), novo.toLowerCase());
		return (existe > 0);
	}
	
}

File: csharp-hfsframework/hfsframework-admin/hfsframework.admin/AdmParametroCategoriaBC.cs
using hfsframework.hfsbase;
using hfsframework.admin.model;
using hfsframework.admin.data;

namespace hfsframework.admin
{
    public class AdmParametroCategoriaBC: BaseBusinessController<AdmParametroCategoria, long, AdmParametroCategoriaRepository>
    {
        private static AdmParametroCategoriaBC instancia = new AdmParametroCategoriaBC();

        public static AdmParametroCategoriaBC Instancia
        {
            get
            {
                return instancia;
            }
        }

        private AdmParametroCategoriaBC():base(new AdmParametroCategoriaRepository())
        {
        }
    }
}
